File: Java-Core/src/main/java/presto/expression/SymbolExpression.java
package presto.expression;

import presto.error.PrestoError;
import presto.error.SyntaxError;
import presto.grammar.Identifier;
import presto.grammar.Symbol;
import presto.runtime.Context;
import presto.type.IType;
import presto.utils.CodeWriter;
import presto.value.IValue;

public class SymbolExpression implements IExpression {

	Identifier name;
	
	public SymbolExpression(Identifier name) {
		this.name = name;
	}

	public Identifier getName() {
		return name;
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		writer.append(name);
	}
	
	@Override
	public IType check(Context context) throws SyntaxError {
		Symbol symbol = context.getRegisteredValue(Symbol.class, name);
		if(symbol==null)
			throw new SyntaxError("Unknown symbol:" + name);
		return symbol.check(context);
	}
	
	@Override
	public IValue interpret(Context context) throws PrestoError {
		Symbol symbol = context.<Symbol>getRegisteredValue(Symbol.class, name);
		if(symbol==null)
			throw new SyntaxError("Unknown symbol:" + name);
		return symbol.interpret(context);			
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/expression/SymbolExpression.js
var SyntaxError = require("../error/SyntaxError").SyntaxError;

function SymbolExpression(name) {
	this.name = name;
	return this;
}

SymbolExpression.prototype.toDialect = function(writer) {
    writer.append(this.name);
};

SymbolExpression.prototype.check = function(context) {
	var symbol = context.getRegisteredValue(this.name);
	if(symbol==null) {
		throw new SyntaxError("Unknown symbol:" + this.name);
	}
	return symbol.check(context);
};

SymbolExpression.prototype.interpret = function(context) {
	var symbol = context.getRegisteredValue(this.name);
	if(symbol==null) {
		throw new SyntaxError("Unknown symbol:" + this.name);
	}
	return symbol.interpret(context);
};

exports.SymbolExpression = SymbolExpression;